source/simulator/core.cc

```diff
@@ -1190,21 +1190,25 @@ namespace aspect
   {
     Amg_preconditioner.reset ();
     Mp_preconditioner.reset ();
-    T_preconditioner.reset ();
-    C_preconditioner.reset ();
```

Comment:
Wouldn't you want to release the memory here?

Reply:
This seems completely unrelated. Why do I need to reset the temperature preconditioner if I reassemble the Stokes preconditioner matrix?

Reply:
Fair enough.